### PR TITLE
Feature/912/change view confirm modal

### PR DESCRIPTION
**Pull request by ArturBekhDEV**
Changed view of Confirm modal, also add functionality to add custom title to button.

### CODE REVIEW

File: src/components/confirm-dialog/ConfirmDialog.styles.ts [hunk 1]
```diff
@@ -4,17 +4,21 @@ export const styles = {
...
-    top: 16,
+    color: 'primary.600',
+    right: '32px',
+    top: '32px',
     p: 0
   },
   title: {
-    p: '13px 16px'
+    color: 'primary.700',
+    typography: { sm: 'h5' },
```
Comment (Tolik170) [inline src/components/confirm-dialog/ConfirmDialog.styles.ts:14]: enum

File: src/components/confirm-dialog/ConfirmDialog.tsx [hunk 2]
```diff
@@ -41,15 +45,15 @@ const ConfirmDialog: FC<ConfirmDialogProps> = ({
       <IconButton onClick={onDismiss} sx={styles.icon}>
         <CloseIcon />
       </IconButton>
-      <DialogContent dividers sx={styles.content}>
+      <DialogContent sx={styles.content}>
         <Typography variant='subtitle1'>{t(message)}</Typography>
       </DialogContent>
       <DialogActions sx={styles.actions}>
         <Button onClick={onConfirm} size='large' variant='tonal'>
```
Comment (FryDay969) [inline src/components/confirm-dialog/ConfirmDialog.tsx:52]: Move styles to styles file.

File: src/components/confirm-dialog/ConfirmDialog.tsx [hunk 3]
```diff
@@ -41,15 +45,15 @@ const ConfirmDialog: FC<ConfirmDialogProps> = ({
       <IconButton onClick={onDismiss} sx={styles.icon}>
         <CloseIcon />
       </IconButton>
-      <DialogContent dividers sx={styles.content}>
+      <DialogContent sx={styles.content}>
         <Typography variant='subtitle1'>{t(message)}</Typography>
```
Comment (FryDay969) [inline src/components/confirm-dialog/ConfirmDialog.tsx:49]: Use TypographyEnum for variant

File: src/components/confirm-dialog/ConfirmDialog.tsx (hunk 2 above)
Comment (Tolik170) [reply src/components/confirm-dialog/ConfirmDialog.tsx:52]: yeah, use variant as typography in styles, and use enums for variants and size everywhere

File: src/components/confirm-dialog/ConfirmDialog.tsx [hunk 4]
```diff
@@ -41,15 +45,15 @@ const ConfirmDialog: FC<ConfirmDialogProps> = ({
...
       </DialogContent>
       <DialogActions sx={styles.actions}>
         <Button onClick={onConfirm} size='large' variant='tonal'>
-          {t('common.yes')}
+          {secondaryBtnTitle ? secondaryBtnTitle : t('common.yes')}
         </Button>
         <Button onClick={onDismiss} size='large' variant='contained'>
-          {t('common.no')}
+          {mainBtnTitle ? mainBtnTitle : t('common.no')}
```
Comment (Tolik170) [inline src/components/confirm-dialog/ConfirmDialog.tsx:56]: I guess, you can set common.no and common.yes as default values in props

File: src/components/confirm-dialog/ConfirmDialog.tsx [hunk 5]
```diff
@@ -41,15 +45,15 @@ const ConfirmDialog: FC<ConfirmDialogProps> = ({
...
-      <DialogContent dividers sx={styles.content}>
+      <DialogContent sx={styles.content}>
         <Typography variant='subtitle1'>{t(message)}</Typography>
       </DialogContent>
       <DialogActions sx={styles.actions}>
         <Button onClick={onConfirm} size='large' variant='tonal'>
-          {t('common.yes')}
+          {secondaryBtnTitle ? secondaryBtnTitle : t('common.yes')}
         </Button>
         <Button onClick={onDismiss} size='large' variant='contained'>
```
Comment (MickhaelSh) [inline src/components/confirm-dialog/ConfirmDialog.tsx:55]: use AppButons instead its large and contained by default

File: src/context/confirm-context.tsx [hunk 6]
```diff
@@ -54,10 +70,12 @@ export const ConfirmationDialogProvider: FC<
   return (
     <ConfirmationDialogContext.Provider value={contextValue}>
       <ConfirmDialog
+        mainBtnTitle={dialogConfig.mainBtnTitle}
```
Comment (MickhaelSh) [inline src/context/confirm-context.tsx:73]: poor naming for buttons one button is for confirm , second for cancel

File: src/components/app-drawer/AppDrawer.tsx [hunk 7]
```diff
@@ -19,12 +20,16 @@ const AppDrawer: FC<AppDrawerProps> = ({
...
+
   const { checkConfirmation } = useConfirm()
 
   const handleCloseDrawer = async () => {
     const confirmed = checkConfirmation({
       message: 'questions.unsavedChanges',
-      title: 'titles.confirmTitle'
+      title: 'titles.confirmTitle',
+      mainBtnTitle: t('common.cancel'),
+      secondaryBtnTitle: t('common.discard')
```
[the remark this answers is not in the document]
Comment (ArturBekhDEV) [reply src/components/app-drawer/AppDrawer.tsx]: as designer said

File: src/components/confirm-dialog/ConfirmDialog.styles.ts (hunk 1 above)
Comment (ArturBekhDEV) [reply src/components/confirm-dialog/ConfirmDialog.styles.ts]: fixed

File: src/components/confirm-dialog/ConfirmDialog.tsx (hunk 2 above)
Comment (ArturBekhDEV) [reply src/components/confirm-dialog/ConfirmDialog.tsx]: fixed

File: src/components/confirm-dialog/ConfirmDialog.tsx (hunk 3 above)
Comment (ArturBekhDEV) [reply src/components/confirm-dialog/ConfirmDialog.tsx]: fixed

File: src/components/confirm-dialog/ConfirmDialog.tsx (hunk 4 above)
Comment (ArturBekhDEV) [reply src/components/confirm-dialog/ConfirmDialog.tsx]: done

File: src/components/confirm-dialog/ConfirmDialog.tsx (hunk 5 above)
Comment (ArturBekhDEV) [reply src/components/confirm-dialog/ConfirmDialog.tsx]: fixed

File: src/components/app-drawer/AppDrawer.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (MickhaelSh) [reply src/components/app-drawer/AppDrawer.tsx]: so what? we should think about what we do, not just do what you were told, it does not have any sense to put buttons cancel and discard together, discuss it with the designer and fix

File: src/hooks/use-confirm.tsx [hunk 8]
```diff
@@ -34,11 +42,19 @@ const useConfirm = (): UseConfirmResult => {
   const checkConfirmation = ({
     message,
     title,
+    confirmBtn,
```
Comment (MickhaelSh) [inline src/hooks/use-confirm.tsx:45]: use the whole words, not "btn", "msg" etc, and you use it as content for the button do so we can put in it not only string for translation but anything we want

File: src/components/app-drawer/AppDrawer.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ArturBekhDEV) [reply src/components/app-drawer/AppDrawer.tsx]: we was discussing with her about it, here is an example that she was providing:
discrad

File: src/components/app-drawer/AppDrawer.tsx [hunk 9]
```diff
@@ -19,12 +20,16 @@ const AppDrawer: FC<AppDrawerProps> = ({
...
+
   const { checkConfirmation } = useConfirm()
 
   const handleCloseDrawer = async () => {
     const confirmed = checkConfirmation({
       message: 'questions.unsavedChanges',
-      title: 'titles.confirmTitle'
+      title: 'titles.confirmTitle',
+      confirmBtn: t('common.discard'),
+      cancelBtn: t('common.cancel')
```
Comment (mxrcury) [inline src/components/app-drawer/AppDrawer.tsx:32]: i am not sure that here we need to use t('') for buttons translations cause you already use it inside ConfirmDialog
image

File: src/components/app-drawer/AppDrawer.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ArturBekhDEV) [reply src/components/app-drawer/AppDrawer.tsx]: discussed and fixed with designer, also changed title and description.

File: src/hooks/use-confirm.tsx (hunk 8 above)
Comment (ArturBekhDEV) [reply src/hooks/use-confirm.tsx]: fixed

File: src/components/app-drawer/AppDrawer.tsx (hunk 9 above)
Comment (ArturBekhDEV) [reply src/components/app-drawer/AppDrawer.tsx]: ye, that's was at last commit, i forgot to take away from confirmdialog translation, right now its fixed

**Review by FryDay969 — APPROVED** [review]

**Review by MickhaelSh — APPROVED** [review]

**Review by mxrcury — APPROVED** [review]

**Review by tsivkadi — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**